tarantool/mesh_connection.py

```diff
@@ -25,21 +35,111 @@ def getnext(self):
+    try:
+        port = int(uri[1])
+    except ValueError:
+        return None
+    
+    if host and port:
+        return {'host': host, 'port': port}
+    else:
+        return None
+
+
 class MeshConnection(Connection):
-    def __init__(self, addrs,
+    '''
+    Represents a connection to a cluster of Tarantool servers.
+
+    This class uses Connection to connect to one of the nodes of the cluster.
+    The initial list of nodes is passed to the constructor in 'addrs' parameter.
+    The class set in 'strategy_class' parameter is used to select a node from
+    the list and switch nodes in case of unavailability of the current node.
+
+    'get_nodes_function_name' param of the constructor sets the name of a stored
+    Lua function used to refresh the list of available nodes. The function takes
+    no parameters and returns a list of strings in format 'host:port'. A generic
+    function for getting the list of nodes looks like this:
+
+    .. code-block:: lua
+
+        function get_cluster_nodes()
+            return {
+                '192.168.0.1:3301',
+                '192.168.0.2:3302',
+                -- ...
+            }
+        end
+
+    You may put in this list whatever you need depending on your cluster
+    topology. Chances are you'll want to make the list of nodes from nodes'
+    replication config. Here is an example for it:
+
+    .. code-block:: lua
+
+        local uri_lib = require('uri')
+
+        function get_cluster_nodes()
+            local nodes = {}
+
+            local replicas = box.cfg.replication
+
+            for i = 1, #replicas do
+                local uri = uri_lib.parse(replicas[i])
+
+                if uri.host and uri.service then
+                    table.insert(nodes, uri.host .. ':' .. uri.service)
+                end
+            end
+
+            -- if your replication config doesn't contain the current node
+            -- you have to add it manually like this:
+            table.insert(nodes, '192.168.0.1:3301')
+
+            return nodes
+        end
+    '''
+
+    def __init__(self, host=None, port=None,
                  user=None,
                  password=None,
                  socket_timeout=SOCKET_TIMEOUT,
                  reconnect_max_attempts=RECONNECT_MAX_ATTEMPTS,
                  reconnect_delay=RECONNECT_DELAY,
                  connect_now=True,
                  encoding=ENCODING_DEFAULT,
-                 strategy_class=RoundRobinStrategy):
-        self.nattempts = 2 * len(addrs) + 1
-        self.strategy = strategy_class(addrs)
-        addr = self.strategy.getnext()
-        host = addr['host']
-        port = addr['port']
+                 call_16=False,
+                 connection_timeout=CONNECTION_TIMEOUT,
+                 addrs=None,
+                 strategy_class=RoundRobinStrategy,
+                 get_nodes_function_name=None,
+                 nodes_refresh_interval=DEFAULT_CLUSTER_DISCOVERY_DELAY_MILLIS):
+
+        addrs_list = []
```

Comment:
It is local variable. Current state of addresses is held in pool

Reply:
Yep, but this does not contradict with my words. When I look at the code first time, I hit this line and ask myself: hey, whether `addrs` is not a list? So I would try to avoid such confusion.

Reply:
To the moon and back. Fixed.